### ISSUE TITLE
coefs fails with lmer in 2.0

### ISSUE THREAD

**Issue by jebyrnes** [jslefche/piecewiseSEM#152]
```
library(lme4)
library(piecewiseSEM)

shipley <- read.csv("../Data/shipley.csv")

# Create list of structural equations
shipley.sem <- psem(
  
  lmer(DD ~ lat + (1|site/tree), na.action = na.omit,
      data = shipley),
  
  lmer(Date ~ DD + (1|site/tree), na.action = na.omit,
      data = shipley),
  
  data = shipley
  
  
  
)

coefs(shipley.sem)

```

produces

```
Error in KRmodcomp.lmerMod(model, reducMod) : 
  Models have either equal fixed mean stucture or are not nested
```

**Comment by jslefche** [comment]
This is a `KRmodcomp` error, not me…such is the issue with using lme4 to get P-values. My advice is to use nlme

____________________________
Jonathan S. Lefcheck, Ph.D.
Post-doctoral Researcher
Bigelow Laboratory for Ocean Science
Phone: +1 (207) 315-2567 ext. 420
Website: www.jonlefcheck.net

________________________________

**Comment by jebyrnes** [comment]
So, I actually do think this is a piecewiseSEM problem. I've been trying to track it. And, here's what's odd. If I set `data = shipley` it now works. Which, to me, means that somewhere there's a scoping issue where it's looking at `data` instead of an object passed through. Haven't found where yet, though.

**Comment by jslefche** [comment]
Yes this has been an longstanding issue. Somehow `GetData` is not working as intended. Wanna take a stab it? Make sure you clone the devel branch now

____________________________
Jonathan S. Lefcheck, Ph.D.
Post-doctoral Researcher
Bigelow Laboratory for Ocean Science
Phone: +1 (207) 315-2567 ext. 420
Website: www.jonlefcheck.net

________________________________

**Comment by jebyrnes** [comment]
Bueno. It's with GetData, eh? Will try and take a look. I tried futzing with some functions downstream, and, yeah, that sounds like it might be the right place. That function ain't never been right!

**Comment by jslefche** [comment]
There's an issue open somewhere with reproducible code…

____________________________
Jonathan S. Lefcheck, Ph.D.
Post-doctoral Researcher
Bigelow Laboratory for Ocean Science
Phone: +1 (207) 315-2567 ext. 420
Website: www.jonlefcheck.net

**Comment by jebyrnes** [comment]
OK, looks like the problem is not GetData, but instead when KRp is called.  The statement that calls it is:

```
   krp <- KRp(i, all.vars_trans(formula(i))[-1], 
            data, intercepts = TRUE)
```

This fails and produces 

```
Error in KRmodcomp.lmerMod(model, reducMod) : 
  Models have either equal fixed mean stucture or are not nested
```

OTOH, if `intercepts=FALSE` no bork.

I have not delved deeply into KRmodcomp - but I do note that the default intercepts setting for KRp is false, despite stdCoefs setting it to TRUE for this call, regardless of what it is otherwise. Is this the problem? What's going on here? Should it be using whatever stdCoefs has its `intercepts` variable set to?

**Comment by jebyrnes** [comment]
Indeed, dug into `KRp` a bit more, and it's the intercepts term that kills it. I think setting `intercepts = intercepts` in the call to `KRp()` in `stdCoefs()` should fix it. Unless there is a reason not to do so, I'll make the fix and push it to devel. Would also love a good example of when `intercepts=TRUE` is the right thing to do.

**Comment by jslefche** [comment]
Added this to devel branch…still hunting down issues with the data argument too

____________________________
Jonathan S. Lefcheck, Ph.D.
Post-doctoral Researcher
Bigelow Laboratory for Ocean Science
Phone: +1 (207) 315-2567 ext. 420
Website: www.jonlefcheck.net

________________________________
From: Jonathan Lefcheck <jlefcheck@bigelow.org>
Sent: Wednesday, June 20, 2018 9:19:03 AM
To: jslefche/piecewiseSEM; jslefche/piecewiseSEM
Cc: Comment
Subject: RE: [jslefche/piecewiseSEM] coefs fails with lmer in 2.0 (#152)

That’s a good solution but doesn’t fix the problem. The intercepts argument is when users want a significance test for the intercept as one might get from `summary.lmerTest` -it refits the model omitting the intercept and then uses `KRmodcomp` to obtain the approximate ddf for the P-value. I think its still a problem in their package but passing `intercepts` from `stdCoefs` should solve it in 95% of cases (since most people don’t want or care about intercepts in SEM)

____________________________
Jonathan S. Lefcheck, Ph.D.
Post-doctoral Researcher
Bigelow Laboratory for Ocean Science
Phone: +1 (207) 315-2567 ext. 420
Website: www.jonlefcheck.net

________________________________

**Comment by jebyrnes** [comment]
Agh - you beat me too it! OK, will pull. Not sure about GetData - it seems to be OK..... this really seems to be the problem. But why intercepts=T is borking is worth looking into. I'm also going to add a lmer test to build.R

**Comment by jslefche** [comment]
Hahaha found it, psem required an argument data= when it should only accept … Fixed and just pushed!

____________________________
Jonathan S. Lefcheck, Ph.D.
Post-doctoral Researcher
Bigelow Laboratory for Ocean Science
Phone: +1 (207) 315-2567 ext. 420
Website: www.jonlefcheck.net

________________________________
